src/main/Port.Adapter/UI/Views/Blazor.Common/TreeView.razor

```diff
@@ -260,78 +260,52 @@
             )
         )
         {
-            if (this.selectedNeuron.IsExpansionTimerEnabled())
-            {
-                this.selectedNeuron.RestartExpansionTimer();
+            this.selectedNeuron.RestartExpansionTimer();
 
-                switch (this.selectedNeuron.CurrentExpansionType)
-                {
-                    case ExpansionType.PostsynapticUntilExternalReferences:
-                        await HandlePostsynapticExpansion();
-                        break;
-                    case ExpansionType.FarthestPresynaptic:
-                        await HandlePresynapticExpansion();
-                        break;
-                }
+            switch (this.selectedNeuron.CurrentExpansionType)
+            {
+                case ExpansionType.PostsynapticUntilExternalReferences:
+                    await HandleExpansion(
+                        Library.Common.RelativeType.Presynaptic, 
+                        Library.Common.RelativeType.Postsynaptic,
+                        n => string.IsNullOrEmpty(n.ExternalReferenceUrl));
+                    break;
+                case ExpansionType.FarthestPresynaptic:
+                    await HandleExpansion(
+                        Library.Common.RelativeType.Postsynaptic, 
+                        Library.Common.RelativeType.Presynaptic,
+                        _ => true);
+                    break;
             }
         }
 
         await base.OnParametersSetAsync();
     }
 
-    private async Task HandlePostsynapticExpansion()
-    {
-        // if children contains selected neuron
-        if (
-            this.Children.Any(
-                x => x.Neuron.Id == this.SelectedNeuron.Neuron.Id && 
-                x.Neuron.Type != Library.Common.RelativeType.Presynaptic && 
-                string.IsNullOrEmpty(SelectedNeuron.Neuron.ExternalReferenceUrl)
-            )
-        )
-        {
-            if (
-                this.SelectedNeuron.Neuron.Type != Library.Common.RelativeType.Presynaptic &&
-                string.IsNullOrEmpty(this.SelectedNeuron.Neuron.ExternalReferenceUrl)
-            )
-            {
-                await InvokeAsync(() => this.SelectedNeuron.Toggle());
-            }
-        }
-        else
-        {
-            foreach (var child in this.Children)
-            {
-                if (this.SelectedNeuron.IsChild(child.Neuron.Id) && 
-                    child.Neuron.Type != Library.Common.RelativeType.Presynaptic && 
-                    string.IsNullOrEmpty(child.Neuron.ExternalReferenceUrl))
-                {
-                    await InvokeAsync(() => child.Toggle());
-                }
-            }
-        }
-    }
-
-    private async Task HandlePresynapticExpansion()
+    private async Task HandleExpansion(
+        Library.Common.RelativeType type, 
+        Library.Common.RelativeType childType,
+        Func<ei8.Cortex.Library.Common.Neuron, bool> otherCondition)
```

Comment:
Please change otherCondition into an optional parameter. ( = null) so that it does not need to be specified on line 284. Then just use (otherCondition == null || otherCondition(...)) in HandleExpansion.